File: ProjectEuler/001-050/S49.cs
using System;
using System.Linq;

namespace ProjectEuler
{
    public class S49 : ISolution
    {
        public string GetAnswer()
        {
            var primes = Utils.GenPrimes(10000).Where(l => l > 1487).Select(Convert.ToInt32).ToArray();
            int length = primes.Length;

            for (int i = 0; i < length - 2; i++)
            {
                for (int j = i + 1; j < length - 1; j++)
                {
                    if (Utils.IsPermutation(primes[i], primes[j]))
                    {
                        for (int k = j + 1; k < length; k++)
                        {
                            if (primes[j] * 2 == primes[i] + primes[k] && Utils.IsPermutation(primes[i], primes[k]))
                            {
                                return string.Format("{0}{1}{2}", primes[i], primes[j], primes[k]);
                            }
                        }
                    }
                }
            }

            return string.Empty;
        }
    }
}

File: src/001-050/S49.cc
#include "S49.h"
#include "../Utils.h"

using namespace std;

namespace ProjectEuler
{
std::string S49::GetAnswer()
{
	auto primes_raw = Utils::GenPrimes(10000);
	std::vector<int> primes;
	for (auto p : primes_raw)
	{
		if (p != 0 && p > 1487)
		{
			primes.push_back((int)p);
		}
	}
	int length = primes.size();

	for (int i = 0; i < length - 2; i++)
	{
		for (int j = i + 1; j < length - 1; j++)
		{
			if (Utils::IsPermutation(primes[i], primes[j]))
			{
				for (int k = j + 1; k < length; k++)
				{
					if (primes[j] * 2 == primes[i] + primes[k] && Utils::IsPermutation(primes[i], primes[k]))
					{
						return std::to_string(primes[i]) + std::to_string(primes[j]) + std::to_string(primes[k]);
					}
				}
			}
		}
	}

	return "";
}

} // namespace ProjectEuler
